app/utility/base_planning_svc.py

```diff
@@ -140,3 +142,21 @@ def _is_fact_bound(fact):
             if not requirement.enforce(combo[0], uf, operation.all_facts()):
                 return False
         return True
+
+    async def _exclude_existing(self, combo, operation):
+        all_hostnames = [agent.paw.split('$')[0].lower() for agent in await self._active_agents(operation)]
+        for item in combo:
+            # prevent backwards lateral movement
+            if 'remote.host' in item.trait:
+                if item.value.split('.')[0].lower() in all_hostnames:
+                    return False
+                elif any(h in item.value.split('.')[0].lower() for h in all_hostnames):
+                    return False
+        return True
+
+    async def _active_agents(self, operation):
```

Comment:
I think this function might be useful in the operation object for use elsewhere. Thoughts?

Reply:
Seems reasonable.